### PR TITLE
Add squid.conf documentation script

### PR DESCRIPTION
**Pull request by yadij**
... circa August 2022.

mk-www-manuals.sh produces version specific configuration manuals
when new code is releasable.

This script takes the version-specific files and updates the website to
publish a unified manual for all Squid versions squid.conf directives.

### CODE REVIEW

File: bin/mk-cfgman-docs.sh
Comment (rousskov) [inline bin/mk-cfgman-docs.sh]: Let's wait for https://github.com/squid-cache/ci/pull/1#pullrequestreview-1066685717 to be resolved. A similar concern applies to this `bin/`-using PR.

**Review by rousskov — DISMISSED** [review]

**Review by kinkie — APPROVED** [review]

File: bin/mk-cfgman-docs.sh
Comment (kinkie) [reply bin/mk-cfgman-docs.sh]: Same consideration as #1 - I don't have an opinion and I don't see it a problem either way.
I would make it a requirement that our scripts use bash. I don't see it as useful to force ourselves to use anachronisms any more than strictly needed

Comment (rousskov) [inline bin/mk-cfgman-docs.sh]: > Francesco:  I would make it a requirement that our scripts use bash. I don't see it as useful to force ourselves to use anachronisms any more than strictly needed

I do not consider the portable/POSIX shell an anachronism; more like an unreachable and poorly defined portability ideal. It certainly has its costs, so I sympathize with your "do not force ourselves" sentiment. On the other hand, using bash-specific features has its costs as well, including higher barriers of entry and fighting about the set of advanced/new bash features that we actually allow.

For this CI repository, I would actually recommend a very relaxed approach when it comes to picking language standards (and many other low-level things!): Let the current primary script author/maintainer use whatever[^1] language (variation) they are most comfortable with, especially when it comes to existing scripts.

[^1]: The usual doc1 and doc2 [principles](https://github.com/measurement-factory/squid-notes/tree/start#documentation-interpretation-principles) apply, of course.

Comment (yadij) [reply bin/mk-cfgman-docs.sh]: FWIW, The portability aspect is somewhat useful on Ubuntu where there are multiple shells present by default. The ```/bin/sh``` links to the more efficient ```dash``` shell and is available at all access levels, so we can be sure our user manual execution of them runs the same as the automation.

Other than that it is just my personal preference to use that minimal syntax for clarity.

Comment (rousskov) [reply bin/mk-cfgman-docs.sh]: > Let's wait for [#1 (review)](https://github.com/squid-cache/ci/pull/1#pullrequestreview-1066685717) to be resolved. A similar concern applies to this `bin/`-using PR.

PR #1 is now using `release/` directory for its script. To speed things up. I am removing my earlier block from this PR under the assumption that this PR will be adjusted to use `release/` before merging.

**Pull request merged**
